client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -149,10 +152,11 @@ const designSetup: Step = function DesignSetup( { navigation, flow } ) {
 
 	const getEventPropsByDesign = ( design: Design ) => ( {
 		theme: design.recipe?.stylesheet,
-		template: design.template,
+		template: design?.template,
 		is_premium: design?.is_premium,
-		flow,
+		flow: showGeneratedDesigns ? 'generated-design-step' : flow,
```

Comment:
I think it's weird to use `step` as the attribute is the `flow`. Maybe we can add a step attribute and keep using the flow.

Reply:
I've added a new property `is_generated` instead of `step` because it seems confusing (at least to me 🥲)